src/Common/CommandArguments.php

```diff
@@ -15,7 +15,35 @@ trait CommandArguments
     protected $arguments = '';
 
     /**
-     * Pass argument to executable. Its value will be automatically escaped.
+     * @var array
+     */
+    protected $argumentsEnv = [];
```

Comment:
Does `$argumentsEnv` have anything to do with environment variables? If this is just the key:value pairs for substitution, then it would probably be better to give it a different name.

Reply:
Env is the name of the argument used by Symfony Process : they pass the placeholders either at exec time, or via environment variables. It's used as environment in this case, since it'd require further changes to the process exec workflow to be passed at the execution step.